linalg/sparsemat.cpp

```diff
@@ -2973,6 +2965,8 @@ SparseMatrix &SparseMatrix::operator=(double a)
 {
    if (Rows == NULL)
    {
+      HostReadWriteI();
+      HostReadWriteJ();
```

Comment:
This seems unnecessary. Is there a reason to want `I` and `J` to match the location of `A`?

Reply:
`done`

Reply:
Moved into `SparseMatrix::AddSubMatrix`

Reply:
`done`